src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/TensorsInfo.cs

```diff
@@ -512,14 +544,11 @@ public TensorInfo(string name, TensorType type, int[] dimension)
 
             public void SetDimension(int[] dimension)
             {
-                if (dimension == null) {
-                    throw new ArgumentException("Max size of the tensor rank is" + Tensor.RankLimit);
-                }
-
                 if (dimension.Length > Tensor.RankLimit) {
                     throw new ArgumentException("Max size of the tensor rank is" + Tensor.RankLimit);
                 }
-                Dimension = (int[])dimension.Clone();
+
```

Comment:
Actually we don't need update SetDimension() method in TensorInfo class, because this is private class and the length of dimension passed to this class always has Tensor.RankLimit.
I suggest updating exception case instead.
```
            public void SetDimension(int[] dimension)
            {
                if (dimension == null) {
                    throw new ArgumentException("The dimension is null, it should be a valid array");
                }

                if (dimension.Length != Tensor.RankLimit) {
                    throw new ArgumentException("The length of the tensor rank should be " + Tensor.RankLimit);
                }
                Dimension = (int[])dimension.Clone();
```

Reply:
Thank you for the suggestion.
Updated it :)